components/wso2is7.key.manager/src/main/java/org/wso2/is7/tenant/management/ISTenantSyncListener.java

```diff
@@ -0,0 +1,451 @@
+    private boolean isTenantSyncEnabled = false;
+    private String identityServerAdminUsername;
+    private String identityServerAdminPassword;
+    boolean isAutoConfigureKeyManagerOfCurrentType = false;
+    boolean skipCreateDefaultResidentKm = false;
+
+    static {
+        apiManagerConfiguration = ServiceReferenceHolder.getInstance()
+                .getAPIManagerConfigurationService().getAPIManagerConfiguration();
+    }
+
+    public ISTenantSyncListener()  {
+        initializeTenantManagementClient();
+    }
+
+     private void initializeTenantManagementClient() {
+        tenantSharingConfiguration = apiManagerConfiguration
+                .getTenantSharingConfiguration(APIConstants.KeyManager.WSO2_IS7_KEY_MANAGER_TYPE);
+
+        skipCreateDefaultResidentKm = Boolean.parseBoolean(apiManagerConfiguration
+                .getFirstProperty(APIConstants.SKIP_CREATE_RESIDENT_KEY_MANAGER));
+
+        if (tenantSharingConfiguration != null && tenantSharingConfiguration.getProperties() != null) {
+            isTenantSyncEnabled = Boolean.parseBoolean(tenantSharingConfiguration.getProperties()
+                    .get(APIConstants.IS7TenantSharingConfigs.ENABLE_TENANT_SYNC));
+            isAutoConfigureKeyManagerOfCurrentType = Boolean.parseBoolean(tenantSharingConfiguration.getProperties()
+                    .get(APIConstants.IS7TenantSharingConfigs.AUTO_CONFIGURE_KEY_MANAGER));
+            identityServerBaseUrl = getRefinedIdentityServerBaseUrl(tenantSharingConfiguration.getProperties()
+                    .get(APIConstants.IS7TenantSharingConfigs.IDENTITY_SERVER_BASE_URL));
+
+            if (isTenantSyncEnabled) {
+                identityServerAdminUsername = tenantSharingConfiguration.getProperties()
+                        .get(APIConstants.IS7TenantSharingConfigs.USERNAME);
+                identityServerAdminPassword = tenantSharingConfiguration.getProperties()
+                        .get(APIConstants.IS7TenantSharingConfigs.PASSWORD);
+                String tenantManagementEndpoint = identityServerBaseUrl + TENANT_MANAGEMENT_API_PATH;
+
+                try {
+                    wso2IS7TenantManagementClient = Feign.builder()
+                            .client(new ApacheFeignHttpClient(APIUtil.getHttpClient(tenantManagementEndpoint)))
+                            .encoder(new GsonEncoder())
+                            .decoder(new GsonDecoder())
+                            .logger(new Slf4jLogger())
+                            .requestInterceptor(new BasicAuthRequestInterceptor(identityServerAdminUsername,
+                                    identityServerAdminPassword))
+                            .errorDecoder(new KMClientErrorDecoder())
+                            .target(WSO2IS7TenantManagementClient.class, tenantManagementEndpoint);
+                } catch (Exception e) {
+                    log.info("Endpoint for tenant management : " + tenantManagementEndpoint);
```

Comment:
lets combine this with the below error message

Reply:
removed that log since it's no longer required